app/services/katello/pulp3/repository/ansible_collection.rb

```diff
@@ -36,7 +36,7 @@ def remote_options
           if root.url.blank?
             super
           else
-            common_remote_options.merge(url: root.url) # whitelist: root.ansible_collection_whitelist)
+            common_remote_options.merge(url: root.url, requirements_file: root.ansible_collection_requirements)
```

Comment:
i'm seeing an issue where if i add a requirements file, save it, then remove the requirements file contents and try to save again we end up sending an empty string, causing an error in pulp.  We probably need to check if the requirements contents are blank and send nil if they are.

Reply:
Updated to send nil as requirement string when blank..